README.md

```diff
@@ -1,9 +1,197 @@
+do that for you.
+
+## Usage
+
+### `uid` and `gid` gotchas
+
+You may leave out the `uid` and `gid` parameters.
+
+**Either always set `uid`, or never set `uid`.** If you set `uid`s for some
+users but not for others, you will inevitably get uid conflicts when you make
+changes. The same is true for groups.
+
+Note that changing a user's `uid` will not update the `uid `on all of that
+user's files. In order to do that, you will need to run something like the
+following on each affected node:
+
+~~~
+# find / -user 1001 -print0 | xargs -0 chown 1034
+~~~
+
+(1001 is the old `uid` and 1034 is the new `uid`.)
+
+`uid` and `gid` are ignored on Windows.
+
+### Passwords
+
+Passwords are handled differently on splatnix (Linux, BSD, etc.) and Windows.
+On splatnix, the password must be in `crypt` format, and on Windows, the
+password must be plain text.
+
+For those reasons it's better to put the passwords in hiera rather than pass
+them directly to the `account::user` resource. Here's an example of a user that
+has the password “hunter2” on both Windows an Linux.
+
+~~~ yaml
+account::user:
+  adam:
+    windows_password: 'hunter2'
+    crypted_password: '$1$QQ8nAVHM$QvthrGVnY2BNu1bJmi7u90'
+~~~
+
+It's better yet to encrypt those values with [Hiera eyaml][]. Using eyaml would
+result in something like the following:
+
+~~~ yaml
+account::user:
+  adam:
+    windows_password: ENC[PKCS7,MIIBeQYJKoZIhvcNAQcDoIIBajCCAWYCA...ghP+fdY1Wd]
+    crypted_password: ENC[PKCS7,MIIBmQYJKoZIhvcNAQcDoIIBijCCAYYCA...SqktJ8Sk8=]
+~~~
+
+Note the accounts must have passwords on Windows. Defining `account::user`
+without a password on Windows will cause that user to be removed.
+
+
+### Predefining home directory files
+
+Some users may have custom configuration they always want in their home
+directory. You can use the `home_source_module` parameter to accomplish that.
+For example, you could add a custom `.bashrc` for `luke` by 1) adding the
+`home_source_module` parameter:
+
+~~~ puppet
+home_source_module => 'profile/users`
+~~~
+
+Then, 2) create a `luke` directory with his `.bashrc` under the `profile::users`
+class, e.g. `site/profile/files/users/luke/.bashrc`.
+
+`home_source_module` does nothing on Windows.
+
+### Shared accounts
+
+`account::user` provides for creating shared accounts, e.g. a `deploy` account
+that multiple users can SSH to.
+
+~~~ puppet
+class profile::users {
+  @account::user { 'gene':
+    uid             => 1002,
+    group           => 'users'
+    groups          => ['admins', 'puppet'],
+    key             => 'AAAAC3NzaC1lZDI1NTE5AAAAIEWkzdlEVmGoAcboiwE16uCuN5dg8xy5mcCPfVnsXMTq',
+    keytype         => 'ssh-ed25519',
+    shared_accounts => ['deploy'],
+  }
+
+  @group { 'deploy': gid => 2000 }
+
+  @account::user { 'deploy':
+    uid   => 2000,
+    group => 'deploy',
+  }
+}
+
+class profile::deploy {
+  realize Account::User['deploy']
+  Ssh::Authorized_key <| tag == "deploy-keys" |>
```

Comment:
Is the tag right here? If so, where does the value `deploy-keys` come from?

Reply:
Ah, good catch; I didn't document that. I updated the copy just above this.